RecoTracker/DeDx/interface/GenericTruncatedAverageDeDxEstimator.h

```diff
@@ -17,7 +17,7 @@ class GenericTruncatedAverageDeDxEstimator : public BaseDeDxEstimator {
     int first = 0, last = Hits.size();
     if (m_fraction > 0) {  // truncate high charge ones
       last -= int(Hits.size() * m_fraction);
-    } else {
+    } else if (m_fraction < 0) {
```

Comment:
I presume this way, you save some additional configuration code, but it looks a bit hack-ish to me. Wouldn't it be better to have a separate configuration flag (asking also the opinion of @cms-sw/reconstruction-l2)

Reply:
indeed, m_fraction==0 having a special meaning can be confusing in the future.

Reply:
but @jpata @mmusich I dont think this is any special meaning!

The meaning of `m_fraction` is "what is the fraction of charges that we want to get rid of, if positive it's the high end, if negative it's the low end", so if the fraction that I want to get rid of is none, then `m_fraction=0` makes a lot of sense, and the original code was just bugged. Dont you agree?

Reply:
Hmm, this is what would happen originally:
https://github.com/cms-sw/cmssw/blob/7ea267110cd0f19eb7dbfac8680676c85b6eccd3/RecoTracker/DeDx/interface/GenericTruncatedAverageDeDxEstimator.h#L21

w/o my modification, the `m_fraction=0` case would go into the conditional where it adds zeros to the `first`, i.e. it would keep it 0, with my modification it will just keep it zero w/o the addition of zeros. So the result is the same, but I thought my modification make it more efficient (no need to do no-op operations)

Reply:
I am not questioning the bug-fix in itself, it's just that after that, the `m_fraction=0` is singled-out and dealt with a specific modifier in the Run 3 case. Perhaps a separate flag for the case in which the truncation should not be done would be more transparent.

Reply:
Fine, is that the preferred solution from @cms-sw/reconstruction-l2 as well given my arguments in https://github.com/cms-sw/cmssw/pull/37387#discussion_r837525346 ?

Reply:
Hi @tvami , I agree with Marco here. I would prefer having a flag that states properly whether the truncation is in place or not

Reply:
Hi @clacaputo yes, I did that in my latest push. That is what actually lead to the current problem "MissingParameter: Parameter 'truncate' not found." which is a bit weird as I added it to the fillDescribtion (and also it worked with the wf I tested this with). I'll have a look at the details in a bit